src/wasm/wasm-binary.cpp

```diff
@@ -132,6 +141,24 @@ void WasmBinaryWriter::finishSection(int32_t start) {
       }
     }
   }
+
+  if (binaryLocationsSizeAtSectionStart != binaryLocations.size()) {
+    // We added the binary locations, adjust them: they must be relative
+    // to the code section.
+    assert(binaryLocationsSizeAtSectionStart == 0);
+    // The actual section start is right before the LEB for the size; we want
+    // to skip over that entire LEB, to get the section body.
+    auto body = start + sizeFieldSize;
+    for (auto& pair : binaryLocations) {
+      // Offsets are relative to the body of the code section: after the
+      // declaration and the size.
```

Comment:
I think "section type" is a better way to refer to that field than "declaration"